src/Octoshift/Services/GithubApi.cs

```diff
@@ -1089,24 +1090,102 @@ public virtual async Task<string> UploadArchiveToGithubStorage(string orgDatabas
 
         if (isMultipart)
         {
-            var url = $"https://uploads.github.com/organizations/{orgDatabaseId.EscapeDataString()}/gei/archive/blobs/uploads";
+            var url = $"{_base_url}/{orgDatabaseId.EscapeDataString()}/gei/archive/blobs/uploads";
 
-#pragma warning disable IDE0028
-            using var multipartFormDataContent = new MultipartFormDataContent();
-            multipartFormDataContent.Add(streamContent, "archive", archiveName);
-#pragma warning restore
-
-            response = await _client.PostAsync(url, multipartFormDataContent);
+            response = await UploadMultipart(archiveContent, archiveName, url);
+            return response;
         }
         else
         {
-            var url = $"https://uploads.github.com/organizations/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
+            var url = $"{_base_url}/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
 
             response = await _client.PostAsync(url, streamContent);
+            var data = JObject.Parse(response);
+            return (string)data["uri"];
         }
+    }
 
-        var data = JObject.Parse(response);
-        return (string)data["uri"];
+    private async Task<string> UploadMultipart(Stream archiveContent, string archiveName, string uploadUrl)
+    {
+        var buffer = new byte[_streamSizeLimit];
+
+        // 1. Start the upload
+        var (startResponse, startHeaders) = await StartUploadAsync(uploadUrl, archiveName, archiveContent.Length);
+
+        var nextUrl = GetNextUrl(startHeaders);
+        var guid = System.Web.HttpUtility.ParseQueryString(nextUrl.Query)["guid"];
+
+        // 2. Upload parts
+        int bytesRead;
+        while ((bytesRead = await archiveContent.ReadAsync(buffer, 0, buffer.Length)) > 0)
+        {
+            nextUrl = await UploadPartAsync(buffer, bytesRead, nextUrl.ToString());
+        }
+
+        // 3. Complete the upload
+        await CompleteUploadAsync(nextUrl.ToString());
+
+        return $"gei://archive/{guid}";
+    }
+
+    private async Task<(string Response, KeyValuePair<string, IEnumerable<string>>[] Headers)> StartUploadAsync(string uploadUrl, string archiveName, long contentSize)
+    {
+        // Create the body for the request
+        var body = new JObject
+        {
+            ["content_type"] = "application/octet-stream",
+            ["name"] = archiveName,
+            ["size"] = contentSize
+        };
+
+        // Post the request and get both response content and headers as a tuple
+        var (response, headers) = await _client.PostTupleAsync(uploadUrl, body);
+
+        // Return the tuple with response and headers
+        return (response, headers);
+    }
+
+
+    private async Task<Uri> UploadPartAsync(byte[] body, int bytesRead, string nextUrl)
+    {
+        var content = new ByteArrayContent(body, 0, bytesRead);
```

Comment:
because we're uploading the entire body and are not using the offset we can use the [other overload](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.bytearraycontent.-ctor?view=net-8.0#system-net-http-bytearraycontent-ctor(system-byte())) that simply accepts the `contents`:
```suggestion
        var content = new ByteArrayContent(body);
```

Reply:
I think we narrowed down that we do need this logic 👍🏽